il.cpp

```diff
@@ -757,6 +757,30 @@ bool GetLowLevelILForInstruction(Architecture* arch, uint64_t addr, LowLevelILFu
 	case ARM64_BLRABZ:
 		il.AddInstruction(il.Call(ILREG(operand1)));
 		break;
+	case ARM64_BFI:
+		il.AddInstruction(il.SetRegister(REGSZ(operand1), REG(operand1),
+			il.Or(REGSZ(operand1),
+				il.And(REGSZ(operand1),
+					il.Const(REGSZ(operand1), ~(((1LL << IMM(operand4)) - 1) << IMM(operand3))),
+					ILREG(operand1)),
+				il.ZeroExtend(REGSZ(operand1), il.ShiftLeft(REGSZ(operand2),
+					il.And(REGSZ(operand2), ILREG(operand2), il.Const(REGSZ(operand2), (1LL << IMM(operand4)) - 1)),
+						il.Const(1, IMM(operand3)))))));
+		break;
+	case ARM64_BFXIL:
+	{
+		// prevent shifting by 64 bits which is UB
```

Comment:
BFI doesn't have this edge case. The only instructions for which width = 64 are:
```
bfxil x2, x3, 0, 0x40
bfi   x2, x3, 0, 0x40
```

Surprisingly, both instructions have the same encoding, and our disassembler breaks the tie with BFXIL.

Should I add this as a comment?